compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -6224,6 +6224,10 @@ protected void visitCheckAndCheckPanicExpr(BLangCheckedExpr checkedExpr, Analyze
             errorTypes.add(exprType);
         }
 
+        if (operatorType.equals(OperatorKind.CHECK.value()) && !data.commonAnalyzerData.errorTypes.empty()) {
```

Comment:
In line 6152, if we make operatorType's type OperatorKind, then we dont need to check the values() and do `equals`, simply == will work.

Reply:
Fixed in [222f18a](https://github.com/ballerina-platform/ballerina-lang/pull/41049/commits/222f18a1126ec66729230c50cdf73bad3b6d46c8), thanks!